File: doc/node.js/client/hello-world.js
var hyperdex_client = require('hyperdex-client')
var c = new hyperdex_client.Client('127.0.0.1', 1982);

function get_callback(obj, err) {
    if (err) { console.log('error ' + err); }
    console.log('get: ' + obj);
}

function put_callback(success, err) {
    if (err) { console.log('error ' + err); }
    console.log('put: ' + success);
    c.get('kv', 'k', get_callback);
}

c.put('kv', 'k', {v: 'Hello World!'}, put_callback);

File: doc/go/client/hello-world.go
package main

import "fmt"
import "github.com/rescrv/HyperDex/bindings/go/client"

func main() {
	c, e, errChan := client.NewClient("127.0.0.1", 1982)
	// permanent error connecting from the cluster
	if e != nil {
		// handle the error here
	}
	// handle transient errors connecting to the cluster
	go func() {
		for e := range errChan {
			// handle the error here
			fmt.Println("error", e)
		}
	}()
	// do the operations
	err := c.Put("kv", "some key", client.Attributes{"v": "Hello World!"})
	if err != nil {
		// handle the error here
	}
	fmt.Println("put: \"Hello World!\"")
	obj, err := c.Get("kv", "some key")
	if err != nil {
		// handle the error here
	}
	fmt.Println("got:", obj)
}
